dspace-server-webapp/src/test/java/org/dspace/app/rest/SubmissionCCLicenseSearchControllerIT.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.junit.Test;
+
+/**
+ * Class to the methods from the SubmissionCCLicenseSearchController
+ * Since the CC Licenses and the corresponding URIs are obtained from the CC License API, a mock service has been
+ * implemented.
+ * This mock service will return a fixed set of CC Licenses using a similar structure to the ones obtained from the
+ * CC License API.
+ * Refer to {@link org.dspace.license.MockCCLicenseConnectorServiceImpl} for more information
+ */
+public class SubmissionCCLicenseSearchControllerIT extends AbstractControllerIntegrationTest {
+
+
+    @Test
+    public void searchRightsByQuestionsTest() throws Exception {
+        getClient().perform(get(
+                "/api/config/submissioncclicenses/search/rightsByQuestions?license=license2&answer_license2-field0" +
+                        "=license2-field0-enum1"))
+                   .andExpect(status().isOk())
+                   .andExpect(jsonPath("$.url", is("mock-license-uri")))
+                   .andExpect(jsonPath("$.type", is("submissioncclicenseUrl")))
+                   .andExpect(jsonPath("$._links.self.href",
+                                       is("http://localhost/api/config/submissioncclicenses/search/rightsByQuestions" +
+                                                  "?license=license2" +
+                                                  "&answer_license2-field0=license2-field0-enum1")));
+    }
+
+    @Test
+    public void searchRightsByQuestionsTestLicenseWithoutFields() throws Exception {
+        getClient().perform(get("/api/config/submissioncclicenses/search/rightsByQuestions?license=license3"))
+                   .andExpect(status().isOk())
```

Comment:
Above I had asked about whether we need to validate at least one answer is provided.  Could we better describe this use case where we can search licenses *without* giving any answers?  Is this to lookup the types of answers allowed?  If so, we should enhance this test to check the response includes info on the types of answers alllowed.

Reply:
Not sure I fully understand the comment here, there are use cases where no answers are needed (as far as I know there is only a single license type that requires answers). What specific tests would you like us to add ?

Reply:
@KevinVdV : What was unclear to me in reviewing this code was exactly what you said above: "there are use cases where no answers are needed".  So, I didn't realize that was a supported use case (as even the name "rightsByQuestions" impied to me that there's always a question and an answer). 

Rather than worrying about any code changes here, could we simply document that use case in our code as an inline comment?  That could just be stating what you said above in the main code...that some licenses actually don't require answers (and you could even link to one or two from the CC API if they are easily available)